tests/installation/usb_install.pm

```diff
@@ -0,0 +1,111 @@
+    assert_script_run("mkdir -p /mnt/ignition");
+    my $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/ignition/config.ign")
+      . " -o /mnt/ignition/config.ign";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    assert_script_run("cat /mnt/ignition/config.ign");
+    save_screenshot;
+
+    # Write combustion script to the same usb with ignition
+    assert_script_run("mkdir -p /mnt/combustion");
+    $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/combustion/script")
+      . " -o /mnt/combustion/script";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    my $SERIALCONSOLE = get_required_var('SERIALCONSOLE');
+    assert_script_run("sed -i 's/SERIALCONSOLE/$SERIALCONSOLE/g' /mnt/combustion/script");
+    assert_script_run("chmod a+x /mnt/combustion/script");
+    assert_script_run("ls -l /mnt/combustion/script && cat /mnt/combustion/script");
+    save_screenshot;
+
+    # Flush to usb stick
+    assert_script_run("sync");
+    assert_script_run("umount -l /mnt");
+    record_info("Ignition and combustion files are successfully downloaded and written to usb $ignition_usb.");
+
+    # Download and dd the iso to the other usb
+    my $download_url = "http://" . get_var('OPENQA_URL', get_var('OPENQA_HOSTNAME')) . "/assets/iso/" . get_required_var('ISO');
+    die "ISO URL is not accessible: $download_url." unless head($download_url);
+    $cmd = "curl -L $download_url | dd of=$iso_usb bs=1M";
+    script_retry($cmd, retry => 2, delay => 10, timeout => 600, die => 1);
+    save_screenshot;
+
+    # Flush
+    assert_script_run("sync");
+    record_info("ISO is saved successfully to usb $iso_usb.");
+
+
+    # Set next boot to usb
+    if (check_var('IPXE_UEFI', '1')) {
+        # Some machines do not support setting floppy boot via ipmitool in uefi boot mode,
+        # so we use efibootmgr instead
+        record_info('efibootmgr output after dd iso to usb:', script_output('efibootmgr'));
+        save_screenshot;
+        # Some machines are problematic which show faulty boot entries as active boot.
+        # Add workaround USB_BOOT_DEV_KEYWORD to filter correct boot.
+        my $cmd = 'efibootmgr | grep usb -i | grep -v generic -i | grep "\*"';
+        $cmd .= get_var('USB_BOOT_DEV_KEYWORD') ? ' | grep ' . get_var('USB_BOOT_DEV_KEYWORD') : '';
+        my $usb_boot = script_output("$cmd");
+        save_screenshot;
+        die "Only 1 bootable USB should be here. But we find in efibootmgr output: $usb_boot." if (!$usb_boot || $usb_boot =~ /\n/);
+        $usb_boot =~ /Boot([0-9A-F]+)\*/m;
+        my $usb_boot_num = $1;
+        assert_script_run("efibootmgr -n $usb_boot_num");
+        save_screenshot;
+        record_info('efibootmgr output after setting next boot to usb:', script_output('efibootmgr'));
+    } else {
+        set_floppy_boot;
```

Comment:
This line is unreachable code.

Reply:
Yes. But if product supports legacy boot, this can be used somehow, so keep it, and it is verified code.

Reply:
There is no ```set_floppy_boo``` definition in this repo. You implemented it somewhere ?

Reply:
Here, https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/da9eef53f51eb7c30dd67b22400d7f8a675cbf41/lib/ipmi_backend_utils.pm#L573